subprojects/core/src/main/java/org/gradle/caching/BuildCacheService.java

```diff
@@ -34,6 +34,16 @@
  *     All other failures will be considered fatal and cause the Gradle build to fail.
  *     Fatal failures could include failing to read or write cache entries due to file permissions, authentication or corruption errors.
  * </p>
+ * <p>
+ *     Every build cache implementation should define a {@link org.gradle.caching.configuration.BuildCache} configuration type.
+ * </p>
+ * <p>
+ *     A build cache implementation constructor <b>must</b> conform to these requirements:
+ *     <ul>
+ *         <li>Must accept an instance of the appropriate {@code BuildCache} configuration object as its first parameter</li>
+ *         <li>Must be annotated with {@link javax.inject.Inject}</li>
+ *     </ul>
+ * </p>
```

Comment:
I don't think you need `@Inject` on the constructor if it only takes the configuration as a parameter.